packages/jupyter-chat/src/selection-watcher.ts

```diff
@@ -207,13 +225,15 @@ function getTextSelection(
     [start, end] = [end, start];
   }
 
+  const language = await languages.getLanguage(editor?.model.mimeType);
   return {
     ...selectionObj,
     start,
     end,
     text,
     numLines: text.split('\n').length,
     widgetId: widget.id,
+    language: language?.name || '',
```

Comment:
If we to follow suggestion, if `language` is `null` it does not need to be returned at all.